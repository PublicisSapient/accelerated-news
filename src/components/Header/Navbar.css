.navbar {
  display: flex;
  align-items: center;
  min-height: var(--px-56);
}

.navbar__brand {
  color: var(--text-primary-default);
  font-size: var(--font-size-lg);
  font-weight: 400;
  margin-right: var(--sp-2);
}

.navbar ul {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

.navbar__link {
  color: var(--text-primary-muted);
  font-weight: 400;
  padding: var(--sp-1);
}

.navbar__link:hover {
  color: var(--neutral-100);
  text-decoration: none;
}

.navbar__link.active {
  color: var(--text-primary-default);
}

.navbar__username {
  display: none;
  color: var(--text-primary-muted);
}
@media (min-width: 460px) {
  .navbar__username {
    display: block;
  }
}

.navbar__signin {
  background-color: var(--neutral-100);
  color: var(--text-default);
}

.navbar__signout {
  color: var(--text-primary-muted);
  margin-left: var(--sp-1);
  cursor: pointer;
}

.navbar__signout:hover {
  color: var(--neutral-100);
}
