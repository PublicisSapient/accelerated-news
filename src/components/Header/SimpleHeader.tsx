import React from 'react';

export const SimpleHeader = () => {
  return (
    <header className="bg-primary-main px-2">
      <nav className="navbar">
        <span className="navbar__brand">Accelerated News</span>
      </nav>
    </header>
  );
};
