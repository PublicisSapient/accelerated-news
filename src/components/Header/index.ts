export * from './Header';
export * from './SimpleHeader';
