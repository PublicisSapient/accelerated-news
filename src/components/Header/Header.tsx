import React from 'react';
import { Navbar } from './Navbar';

export const Header = () => {
  return (
    <header className="bg-primary-main">
      <div className="container">
        <Navbar />
      </div>
    </header>
  );
};
