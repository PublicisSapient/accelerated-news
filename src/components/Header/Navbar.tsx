import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';

export const Navbar = () => {
  return (
    <nav className="navbar">
      <span className="navbar__brand mobile">News</span>
      <span className="navbar__brand desktop">Accelerated News</span>
      <ul>
        <li>
          <NavLink className="navbar__link" to="/" end>
            Headlines
          </NavLink>
        </li>
        <li>
          <NavLink className="navbar__link" to="/sports" end>
            Sports
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
