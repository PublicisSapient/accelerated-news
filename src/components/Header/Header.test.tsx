import React from 'react';
import { render } from '../../test/test-utils';
import { Header } from './Header';

describe('<Header />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
