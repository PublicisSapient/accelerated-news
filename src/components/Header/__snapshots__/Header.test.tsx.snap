// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<DocumentFragment>
  <header
    class="bg-primary-main"
  >
    <div
      class="container"
    >
      <nav
        class="navbar"
      >
        <span
          class="navbar__brand mobile"
        >
          News
        </span>
        <span
          class="navbar__brand desktop"
        >
          Accelerated News
        </span>
        <ul>
          <li>
            <a
              aria-current="page"
              class="navbar__link active"
              href="/"
            >
              Headlines
            </a>
          </li>
          <li>
            <a
              class="navbar__link"
              href="/sports"
            >
              Sports
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
